luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+import logging
+import pkg_resources
+import sys
+from typing import List
+import matplotlib.pyplot as plt
+
+import luigi
+import hail as hl
+
+from luigi_pipeline.lib.hail_tasks import (
+    HailMatrixTableTask,
+    GCSorLocalTarget,
+)
+from luigi_pipeline.seqr_loading import (
+    check_if_path_exists,
+    contig_check,
+    SeqrValidationError,
+)
+
+logger = logging.getLogger(__name__)
+
+PYSPARK_SUBMIT_ARGS = "--driver-memory 400G pyspark-shell"
+
+FEMALE_PLOIDY = "XX"
+MALE_PLOIDY = "XY"
+SEX_PLOIDY_ANNOTATIONS = [
+    "is_female",
+    "f_stat",
+    "n_called",
+    "expected_homs",
+    "observed_homs",
+    "sex",
+]
+GRCh37_STANDARD_CONTIGS = {
+    "1",
+    "10",
+    "11",
+    "12",
+    "13",
+    "14",
+    "15",
+    "16",
+    "17",
+    "18",
+    "19",
+    "2",
+    "20",
+    "21",
+    "22",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+    "X",
+    "Y",
+    "MT",
+}
+GRCh38_STANDARD_CONTIGS = {
+    "chr1",
+    "chr10",
+    "chr11",
+    "chr12",
+    "chr13",
+    "chr14",
+    "chr15",
+    "chr16",
+    "chr17",
+    "chr18",
+    "chr19",
+    "chr2",
+    "chr20",
+    "chr21",
+    "chr22",
+    "chr3",
+    "chr4",
+    "chr5",
+    "chr6",
+    "chr7",
+    "chr8",
+    "chr9",
+    "chrX",
+    "chrY",
+    "chrM",
+}
+OPTIONAL_CHROMOSOMES = ["MT", "chrM", "Y", "chrY"]
+VARIANT_THRESHOLD = 100
+
+
+class SexPloidyCheckTask(luigi.Task):
+    """
+    Inherits from a Hail MT Class to get helper function logic. Main logic to do annotations here.
+    """
+
+    source_paths = luigi.Parameter(
+        default="[]", description="Path or list of paths of VCFs to be loaded."
+    )
+    wes_filter_source_paths = luigi.OptionalParameter(
```

Comment:
Should this default to "[]" so we can iterate over it without a `None` check later?